zomes/blog/code/src/main_capability.rs

```diff
@@ -3,6 +3,23 @@ use hdk::globals::G_MEM_STACK;
 use serde_json;
 
 zome_functions! {
+    check_sum: |num1: u32, num2: u32| {
+        #[derive(Serialize)]
+        struct SumInput {
+            num1: u32,
+            num2: u32,
+        };
+        let call_input = SumInput {
+            num1: num1,
+            num2: num2,
+        };
+        let maybe_result = hdk::call("summer", "main", "sum", serde_json::to_value(call_input).unwrap());
+        match maybe_result {
```

Comment:
Looks great! I would also look into syntax like question marks or unwrap_or or map. They can drastically reduce the lines of code.